### ISSUE TITLE
Upgrades fail with 'No X with the name Y found', despite there existing X objects with Y name

### ISSUE THREAD

**Issue by yuvipanda** [helm/helm#3785]
Output of `helm version`:

```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```

Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.2", GitCommit:"bdaeafa71f6c7c04636251031f93464384d54963", GitTreeState:"clean", BuildDate:"2017-10-24T19:48:57Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.4-gke.1", GitCommit:"10e47a740d0036a4964280bd663c8500da58e3aa", GitTreeState:"clean", BuildDate:"2018-03-13T18:00:36Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
GKE, 1.9.4+

When running helm upgrade, I now frequently get errors like:

```
Error: UPGRADE FAILED: no ConfigMap with the name "egress-proxy-config" found
```

However, on inspecting the namespace, you'll see that the configmap *does* exist:

```
23:27 $ kubectl --context=gke_data8x-scratch_us-central1_staging-cluster-alpha --namespace=hub-alpha-01  get cm
NAME                  DATA      AGE
egress-proxy-config   1         15m
hub-config            33        5d
```

In the tiller logs, I see:

```
[tiller] 2018/03/29 06:24:48 warning: Upgrade "hub-alpha-01" failed: no ConfigMap with the name "egress-proxy-config" found
```

If I manually delete the configmap and re-run helm upgrade, it works. 

I've run into this problem with configmap, secret and service account objects so far. They seem to occur randomly, but get stuck on a release once they do occur. This has screwed our CD pipeline, unfortunately :(

**Comment by jacobtomlinson** [comment]
In #3791 I also had the same issues but with service accounts and roles.

**Comment by minrk** [comment]
I ran into this upgrading helm from 2.7.2 to 2.8.2 and became suspicious that it might be related to #2941 - multiple deployments marked as deployed, because `helm ls` showed multiple entries for the same deployment. After upgrading helm to 2.8.2, it no longer showed these, but sure enough in the tiller logs:

```
[kube] 2018/04/30 08:36:33 building resources from updated manifest
[kube] 2018/04/30 08:36:33 checking 80 resources for changes
[kube] 2018/04/30 08:36:33 Looks like there are no changes for Secret "hub-secret"
[kube] 2018/04/30 08:36:33 Looks like there are no changes for Secret "binder-secret"
[kube] 2018/04/30 08:36:33 Looks like there are no changes for ConfigMap "staging-grafana-dashs"
[kube] 2018/04/30 08:36:33 Looks like there are no changes for ConfigMap "staging-nginx-ingress-controller"
[tiller] 2018/04/30 08:36:33 warning: Upgrade "staging" failed: no ConfigMap with the name "proxy-patches-nginx-config" found
[storage] 2018/04/30 08:36:33 updating release "staging.v271"
[storage] 2018/04/30 08:36:33 updating release "staging.v419"
```

After running

```
kubectl edit configmap --namespace=kube-system staging.v271
```

to set `status: SUPERSEDED` (it was incorrectly marked as DEPLOYED) and running the upgrade again, it failed in the same way. On looking at the logs, it did the same thing, but with another duplicate release. Repeating twice, fixing the remaining configmaps for the wrong-state deployments resulted in a successful deploy.

So I think there's still something not quite right in 2.8.2's handling of the unclean state where multiple revisions have been marked as deployed, but at least cleaning up that state seems to resolve the issue.

**Comment by gintautassulskus** [comment]
I am getting the same issue with 2.9.1 despite the existing deployment running in that namespace.
```
Error: UPGRADE FAILED: no Namespace with the name "foo" found
```

**Comment by omerlh** [comment]
Is there an update on this issue? I'm experiencing the same issue...

**Comment by jstriebel** [comment]
We are experiencing this issue, too.

It seems to happen with different types of resources (seen pods, services, ingresses), but for us only in one specific release, with two specific characteristics:
* the release is is sometimes upgraded in parallel,
* it has a `post-install,post-upgrade` hook (a job).

The failure then blocks any subsequent upgrades, also on other releases. Not sure if #4223 might workaround this issue.

Helm version is `v2.9.0`

**Comment by alexandersauerbier** [comment]
+1 (happens with helm v2.8.2 and ConfigMap ... same example as the issue-starter had. and, with me also `kubectl get configmaps <map_name> --namespace <namespace_name>` gives the correct output)

**Comment by jpaulgs** [comment]
Is there an update here?

I've encountered this a couple of times with 2.8.1, The first time it was caused because pods had failed to be deployed by kubernetes (no node had enough resources).  However I've encountered this error again today with no obvious cause.

```
Client: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
```

**Comment by bacongobbler** [comment]
I believe this is a duplicate of https://github.com/kubernetes/helm/issues/1193, of which I *think* I've managed to find the behavior that causes this bug. Haven't heard of any good solutions to the problem so far, though.

**Comment by bacongobbler** [comment]
long write-up on the bug's behavior and the attempted solutions can be found here: https://github.com/kubernetes/helm/pull/4223#issuecomment-397413568

**Comment by bacongobbler** [comment]
dupe of #1193

**Comment by DimitrisD** [comment]
> Hi @DimitrisD, do you mind share How exactly you delete the ClusterRoleBinding? Thanks!

You can get all the clusterrolebindings and delete them using kubectl. Eg:
```kubectl get clusterrolebindings --all-namespaces``` to get the one you need to delete, then use ```kubectl delete clusterrolebinding <your id> -n <your namespace>```

**Comment by usefree** [comment]
--force option helped for me
